### DISCUSSION TITLE
Include `Symbol` in `ContentValue`

### DISCUSSION THREAD

**Discussion by gilest** [typed-ember/glint#611] [Ideas]
Interested to hear thoughts about this.

```js
const example = Symbol('example');

<template>
  {{example}}
</template>
```

Will render 

```
Symbol(example)
```

**Comment by dfreeman** [comment]
You can also write this:

```tsx
const example = ['hello', 'world'];

<template>
  {{example}}
</template>
```

And render this:
```
hello,world
```

The reason we prohibit both of those cases isn't that you'll get a runtime exception when trying to render the value, it's that values of those types don't typically represent user-consumable information. 9 times out of 10 if we encounter a symbol or object being used directly as template content, it represents a typo or refactoring mistake like accidentally referencing source data instead of the human-formatted version.

TS's types for native language features are full of similar cases to this—the `Symbol` constructor is a great example. I can write `Symbol(['hello', 'world'])` and that will coerce the array to a string at runtime without complaint, but TS still flags it as a type error because it represents a probable mistake.

For cases where you do want to coerce an arbitrary value to a string no matter what it is, you can either coerce it in TS and reference the result in the template, or use the `concat` helper to implicitly perform the coercion for you.

```tsx
import { concat } from '@ember/helper';

const example = Symbol('example');
const exampleString = example.toString();

// Both of these will render `Symbol(example)` with no type error:
<template>
  {{exampleString}}
  {{concat example}}
</template>
```

**Reply by gilest** [reply]
Thanks @dfreeman. Adding Glint types is quite a good shakedown of a public API in this case
